app/models/sac_cas/role/active_membership_validations.rb

```diff
@@ -8,18 +8,33 @@
 module SacCas::Role::ActiveMembershipValidations
   extend ActiveSupport::Concern
 
-  MITGLIED_ROLES = [Group::SektionsMitglieder::Mitglied,
-    Group::SektionsMitglieder::MitgliedZusatzsektion]
-
   included do
     validate :assert_has_active_membership_role
   end
 
   def assert_has_active_membership_role
-    unless Role.with_inactive.exists?(type: MITGLIED_ROLES.map(&:sti_name),
-      person_id: person_id,
-      group_id: group_id)
-      errors.add(:person, :must_have_mitglied_role_in_group)
+    # Only validate if both dates are set (otherwise the role will be invalid anyway)
+    return unless start_on && end_on
+
+    memberships = Role.with_inactive.where(type: SacCas::MITGLIED_ROLES.map(&:sti_name), person_id:, group_id:)
+
+    # Find open ended membership and check coverage (early return)
+    openended_membership = memberships.find_by(end_on: nil)
+    return if openended_membership.present? && openended_membership.start_on <= start_on
+
+    # If the end_on date is present, we can iterate over the active period and check all days.
+    # Otherwise we have to check the active period until the start of the open ended membership.
+    # Unless there is no open ended membership, in which case it can not be covered.
+    check_range = end_on.present? ? active_period : start_on...openended_membership&.start_on
```

Comment:
Das muss ja aufgrund Zeile 17 sowieso immer `active_period` sein oder?

Reply:
stimmt, dann kann man das in https://github.com/hitobito/hitobito_sac_cas/blob/master/app/models/sac_cas/role/mitglied_zusatzsektion.rb#L35 auch gleich entfernen